### ISSUE TITLE
Confusion about "Average Accuracy"

### ISSUE THREAD

**Issue by serdaryildiz** [WindVChen/Push-and-Pull-Network#2]
Hello, 

Thank you for your research. I'm not sure what "Average Accuracy" means. The validation function computes average accuracy as follows: 

    classesCal = [0 for x in range(args.num_classes)]
    correctCal = [0 for x in range(args.num_classes)]
    model.eval()
    accNum = 0
    accAll = 0
    lossAll = 0
    for image, label in tqdm.tqdm(valid_loader):
        image = image.cuda()
        label = label.cuda()
        if not args.baseline:
            with torch.no_grad():
                t1, t2, _, _, _, _, _, _, _ = model(image, image, args.baseline)
            loss = (classify_criterion(t1, label) + classify_criterion(t2, label)) * 0.5
        else:
            with torch.no_grad():
                t1, _, _, _, _, _, _, _ = model(image, None, args.baseline)
            loss = classify_criterion(t1, label)
        _, prediction = torch.max(t1, 1)
        accAll += (prediction == label).sum().item()
        for i, j in enumerate(label):
            classesCal[j.item()] += 1
            if (prediction[i] == label[i]):
                correctCal[j.item()] += 1
        lossAll += loss.item() * image.size(0)
    for i in range(args.num_classes):
        if (classesCal[i]):
            accNum += correctCal[i] / classesCal[i]
            print("class {} \t acc: {}".format(i, correctCal[i] / classesCal[i]))
    accNum /= args.num_classes
    print("lossAvg: {} \t avgAcc: {} \t allAcc: {}".format(lossAll / len(valid_loader.dataset), accNum,
                                                           accAll / len(valid_loader.dataset))),`

Is this calculation average recall, or did I miss something?

thanks for your time.

**Comment by WindVChen** [comment]
Hi @serdaryildiz ,

Thanks for your attention to our work.

Yes. Average Accuracy means the average of each category's recall. You can refer to Section IV-C of our paper for more details.
